rust/src/lib.rs

```diff
@@ -7,8 +7,140 @@
 //
 // On the date above, in accordance with the Business Source License, use of
 // this software will be governed by the GNU Lesser General Public License v3.
-#![allow(unused)]
+
+use std::mem::MaybeUninit;
+
+use crate::{error::Error, types::*};
 
 mod error;
 mod ffi;
 mod types;
+
+/// Opens a new [CarmenS6Db] database object based on the provided implementation maintaining
+/// its data in the given directory. If the directory does not exist, it is
+/// created. If it is empty, a new, empty state is initialized. If it contains
+/// state information, the information is loaded.
+pub fn open_carmen_s6_db(
+    _schema: u8,
```

Comment:
The `_schema` parameter actually determines what "sX" version should be produced. So this function is intended to be used for all kind of different schemas. I would thus suggest to
- rename the function to `open_carmen_db`
- rename the type `CarmenS6Db` trait to `CarmenDb` since the trait contains nothing S6 specific and the FFI is in general schema-independent

If there is no support for a given schema on the Rust side (schemas 1-5, >=7), an error should be returned.

Wdyt?

Reply:
Sounds good